src/nspawn/nspawn.c

```diff
@@ -1074,6 +1074,15 @@ static int parse_argv(int argc, char *argv[]) {
         if (getenv_bool("SYSTEMD_NSPAWN_MOUNT_RW") <= 0)
                 mount_settings |= MOUNT_RO;
 
+        if (arg_userns_mode != USER_NAMESPACE_NO)
+                mount_settings |= MOUNT_USE_USERNS;
+
+        if (arg_private_network)
+                mount_settings |= MOUNT_USE_NETNS;
+
+        if (getenv_bool("SYSTEMD_NSPAWN_USE_NETNS") <= 0)
+                mount_settings |= MOUNT_USE_NETNS;
```

Comment:
not sure I get this one. What's the precise usecase here? Why wouldn't one use --private-network here?